flink-runtime/src/main/java/org/apache/flink/runtime/state/AbstractKeyedStateBackend.java

```diff
@@ -348,7 +349,7 @@ public int numKeyValueStatesByName() {
     }
 
     // TODO remove this once heap-based timers are working with RocksDB incremental snapshots!
-    public boolean requiresLegacySynchronousTimerSnapshots() {
+    public boolean requiresLegacySynchronousTimerSnapshots(CheckpointType checkpointOptions) {
```

Comment:
The parameter seems to never be used?

Reply:
Ah, that's a mistake! Sorry, I am fixing it and adding a test for it.